### PR TITLE
Added docker v2 search using catalog. Defaulting to v2 search.

### PR DESCRIPTION
**Pull request by ankanm**
* Added v2 search using catalog.
* On error trying v1 search.

@jordmoz

### CODE REVIEW

File: cli/dockerutils.py [hunk 1]
```diff
@@ -32,3 +34,28 @@ def docker_build(self, image_tag, docker_file):
...
+
+    def docker_search(self, registry, name, application):
+        result = ""
+        try:
+            result = self.docker_search_v2(registry, name, application)
+        except (Exception) as e:
+            print("The following error occurred when attempting search using docker v2 catalog: %s" %
+                  e, file=sys.stderr)
+            print("Attempting docker v1 search")
+            result = self.docker_search_v1(registry, name, application)
```
Comment (04msambit) [inline cli/dockerutils.py]: What if the docker_search_v1 throws an exception ?. Should we not catch it here and decide result accordingly.

File: cli/dockerutils.py [hunk 2]
```diff
@@ -32,3 +34,28 @@ def docker_build(self, image_tag, docker_file):
     def docker_push(self, image):
         exit_code = os.system("docker push {0}".format(image))
         return exit_code
+
+    def docker_search_v1(self, registry, name, application):
+        result = subprocess.check_output("docker search {0}/{1}-{2}".format(
```
Comment (04msambit) [inline cli/dockerutils.py]: We should put the code in try..except block to handle if at all any exception occurs. As well as "raise" it.

**Comment by 04msambit** [comment]
LGTM.

Verification:

- Get the specific branch using install <branch> in roger-cli. Able to deploy app.
- Switch back to master , roger deploy fails. 

The fix seems to be working.

**Comment by jordmoz** [comment]
LGTM

**Pull request merged**